syntax = "proto3";

package ocr;

// 请求消息包含PDF文件的字节数据、起始页和结束页
message OCRRequest {
  bytes pdf_data = 1;      // PDF文件的二进制数据
  int32 start_page = 2;    // 起始页（0基）
  int32 end_page = 3;      // 结束页（0基）
}

// 单页的OCR结果
message PageOCRResult {
  int32 page_number = 1;   // 页码
  string text = 2;         // OCR提取的文本
  string llm_response = 3; // LLM处理后的响应（可选）
}

// 响应消息包含多个页的OCR结果
message OCRResponse {
  repeated PageOCRResult results = 1; // 多页OCR结果列表
}

// 定义OCR服务
service OCRService {
  rpc AnalyzePDF (OCRRequest) returns (OCRResponse);
}
